doc/applications.md

```diff
@@ -0,0 +1,151 @@
+
+```js
+// ES2015
+const increaseButton = document.querySelector('#increase');
+const increase$ = Rx.Observable.fromEvent(increaseButton, 'click')
+  .map(() => state => {...state, count: state.count + 1});
+
+// We create an object with our state. Whenever a new state change function
+// is received we call it and pass the state. The new state is returned and
+// ready to be changed again on the next click
+const state$ = increase$.scan((state, changeFn) => changeFn(state), {count: 0});
+```
+
+We can now add a a couple of more observables which will also change the same state store.
+
+```js
+const increaseButton = document.querySelector('#increase');
+const increase$ = Rx.Observable.fromEvent(increaseButton, 'click')
+  .map(() => state => {...state, count: state.count + 1});
+
+const decreaseButton = document.querySelector('#decrease');
+const decrease$ = Rx.Observable.fromEvent(decreaseButton, 'click')
+  .map(() => state => {...state, count: state.count - 1});
+
+const input = document.querySelector('#input');
+const input$ = Rx.Observable.fromEvent(input, 'keypress')
+  .map(event => state => {...state, inputValue: event.target.value});
+
+// We merge the three state change producing observables
+const state$ = Rx.Observable.merge(
+  increase$,
+  decrease$,
+  input$
+).scan((state, changeFn) => changeFn(state), {
+  count: 0,
+  inputValue: ''
+});
+
+// We subscribe to state changes and update the DOM
+state$.subscribe((state) => {
+  document.querySelector('#count').innerHTML = state.count;
+  document.querySelector('#hello').innerHTML = 'Hello ' + state.inputValue;
+});
+```
+
+We can take the state store approach and use it with many different frameworks and libraries.
+
+### Immutable JS
+You can also create a global state store for your application using [Immutable JS](https://facebook.github.io/immutable-js/). Immutable JS is a great way to create immutable state stores that allows you to optimize rendering by doing shallow checks on changed values.
+
+```js
+import Immutable from 'immutable';
+import someObservable$ from './someObservable';
+import someOtherObservable$ from './someOtherObservable';
+
+const initialState = {
+  foo: 'bar'
+};
+
+const state$ = Observable.merge(
+  someObservable$,
+  someOtherObservable$
+).scan((state, changeFn) => changeFn(state), Immutable.fromJS(initialState));
+
+export default state$;
+```
+
+Now you can import your state in whatever UI layer you are using.
+
+```js
+import state$ from './state';
+
+state$.subscribe(state => {
+  document.querySelector('#text').innerHTML = state.get('foo');
+});
+```
+
+### React
+To reduce boilerplate you can use [rxjs-react-component](https://www.npmjs.com/package/rxjs-react-component). It will allow you to expose observables that maps to state changes, as if you would do `this.setState({count: 1})`.
```

Comment:
Yeah, good point! Let me create a more generic example and rather list some tools that can help you work with observables more effectively in React. The `rxjs-react-component` thingy was a thing of inspiration from my end. I like the idea that observables can be this thing between interaction events and producing state. Have some ideas to make the API even smoother on that small lib, but yes... let me work on that on the side and list it with other helpful tools as well :)